test/cppsuite/src/main/database_operation.cpp

```diff
@@ -124,6 +124,20 @@ database_operation::populate(
     logger::log_msg(LOG_INFO, "Populate: finished.");
 }
 
+void
+database_operation::background_compact_operation(thread_worker *tc)
+{
+    logger::log_msg(
+      LOG_ERROR, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+    /* This needs to be executed only once in the workload. */
+    if (tc->running()) {
```

Comment:
Is it possible for this function to execute when `tc->running()` is false? If it is then there's the risk of a race here. If not then we could drop the `if`

Reply:
No, it cannot be executed if the test has ended unless we are not using the framework correctly which should not be the case.
Fixed in 6e41ce95d2d5e7cc04e9c56989fa795417efab6d.